### PR TITLE
Force all developer application redirect URIs to use HTTPS

### PR DESCRIPTION
**Pull request by aboutdavid**
Should also fix [this sentry issue](https://obl-ong.sentry.io/issues/5060213714/events/b8be1bf3f1a74952913aa8750d00eea4/)
> Validation failed: Redirect URI must be an HTTPS/SSL URI. (ActiveRecord::RecordInvalid)

### CODE REVIEW

**Comment by aboutdavid** [comment]
Closes #146

File: app/controllers/developers/applications_controller.rb [hunk 1]
```diff
@@ -1,20 +1,18 @@
 class Developers::ApplicationsController < ApplicationController
-  nested_layouts "layouts/admin"
+  nested_layouts 'layouts/admin'
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:2]: this will fail linting

File: app/controllers/developers/applications_controller.rb [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
 
   before_action do
     @developers = true
   end
 
-  before_action except: [:index, :request, :provision, :create] do
+  before_action except: %i[index request provision create] do
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:8]: what's the rationale behind this change, feels less descriptive imo?

File: app/controllers/developers/applications_controller.rb [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
     if (current_application.provisional? || current_application.owner_id != current_user.id) && !current_user.admin?
-      render plain: "403 Forbidden or Provisional Domain", status: 403
+      render plain: '403 Forbidden or Provisional Domain', status: 403
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:10]: same thing - will also fail linting

File: app/controllers/developers/applications_controller.rb [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
     end
   end
 
   before_action only: [:create] do
-    if !current_user.admin?
-      render plain: "403 Forbidden", status: 403
-    end
+    render plain: '403 Forbidden', status: 403 unless current_user.admin?
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:15]: again

File: app/controllers/developers/applications_controller.rb [hunk 5]
```diff
@@ -44,10 +42,16 @@ def destroy_scope
     scopes = @application.scopes.to_a
     scopes.delete(params[:scope])
     @application.update!(scopes: Doorkeeper::OAuth::Scopes.from_array(scopes))
-    redirect_back(fallback_location: developers_applications_path(id: params[:id]), notice: "Destroyed scope #{params[:scope]}")
+    redirect_back(fallback_location: developers_applications_path(id: params[:id]),
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:45]: why change

File: app/controllers/developers/applications_controller.rb [hunk 6]
```diff
@@ -56,7 +60,7 @@ def add_redirect_uri
     rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique => e
       flash.notice = e.message
     else
-      flash.notice = "Added Redirect URI"
+      flash.notice = 'Added Redirect URI'
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:63]: same

File: app/controllers/developers/applications_controller.rb [hunk 7]
```diff
@@ -67,14 +71,14 @@ def destroy_redirect_uri
     uris = @application.redirect_uri.split("\r\n")
     uris.delete(params[:redirect_uri])
     @application.update!(redirect_uri: uris.join("\r\n"))
-    redirect_back(fallback_location: developers_applications_path(id: params[:id]), notice: "Destroyed Redirect URI")
+    redirect_back(fallback_location: developers_applications_path(id: params[:id]), notice: 'Destroyed Redirect URI')
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:74]: same

File: app/controllers/developers/applications_controller.rb [hunk 8]
```diff
@@ -67,14 +71,14 @@ def destroy_redirect_uri
...
   end
 
   def update
     @application = current_application
     @application.update!(name: params[:name]) if params[:name]
     @application.update!(confidential: params[:confidential].to_i.zero?) if params[:confidential]
-    redirect_back(fallback_location: developers_applications_path(id: params[:id]), notice: "Updated application")
+    redirect_back(fallback_location: developers_applications_path(id: params[:id]), notice: 'Updated application')
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:81]: same

File: app/controllers/developers/applications_controller.rb [hunk 9]
```diff
@@ -83,14 +87,22 @@ def destroy
   end
 
   def create
-    @application = Doorkeeper::Application.new(name: params[:name], redirect_uri: params[:redirect_uri], confidential: true)
+    @application = Doorkeeper::Application.new(name: params[:name], redirect_uri: params[:redirect_uri],
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:90]: why change

File: app/controllers/developers/applications_controller.rb [hunk 10]
```diff
@@ -83,14 +87,22 @@ def destroy
...
   def provision
-    @application = Doorkeeper::Application.new(name: params[:name], redirect_uri: params[:redirect_uri], plan: params[:plan], confidential: true, provisional: true)
+    uri = URI.parse(params[:redirect_uri])
+    if uri.scheme != 'https'
+      redirect_back(fallback_location: developers_applications_path, notice: 'URIs must use HTTPS')
+      return
+    end
+
+    @application = Doorkeeper::Application.new(name: params[:name], redirect_uri: params[:redirect_uri],
+                                               plan: params[:plan], confidential: true, provisional: true)
```
Comment (reesericci) [inline app/controllers/developers/applications_controller.rb:105]: weird line break

File: app/views/developers/applications/index.html.erb
Comment (reesericci) [inline app/views/developers/applications/index.html.erb:1]: the flashes are in layouts - not in individual views, so update it there

File: app/views/developers/applications/request.html.erb [hunk 11]
```diff
@@ -1,54 +1,40 @@
 <h1 class="font-heading text-3 lg:text-4 xl:text-5">Request an application</h1>
 <br><br>
-
+<% flash.each do |type, msg| %>
```
Comment (reesericci) [inline app/views/developers/applications/request.html.erb:3]: same thing - flash in layout

File: app/views/developers/applications/request.html.erb [hunk 12]
```diff
@@ -1,54 +1,40 @@
...
-        min-width: 450px;
-        min-height: 10rem;
-    }
-
-    a {
-        color: var(--winter-sky);
-    }
-<% end %>
\ No newline at end of file
+  form { max-width: 50vw; } .domain { font-size: 2rem; } label { font-size:
```
Comment (reesericci) [inline app/views/developers/applications/request.html.erb:35]: let's not minify css

**Review by reesericci — APPROVED** [review]
Overall the approach LGTM - just run the linter & fix some weird formatting

**Review by reesericci — CHANGES_REQUESTED** [review]
requesting changes

File: app/controllers/developers/applications_controller.rb (hunk 2 above)
Comment (aboutdavid) [reply app/controllers/developers/applications_controller.rb:8]: uhhhh the weird formatter did that

**Review by reesericci — APPROVED** [review]
LGTM!
